scripts/pipeline/orthology_benchmark.py

```diff
@@ -157,12 +160,49 @@ def get_core_names(species_names: List[str], host: str, port: int, user: str) ->
     return core_names
 
 
-def prep_input_for_orth_tools():
-    """Docstring"""
+def prep_input_for_orth_tools(source_dir: str, target_dir_ofinder: str) -> None:
+    """Prepares input files for selected orthology inference tools.
 
+    For OrthoFinder: Creates symlinks to input fasta files in `target_dir_ofinder`.
+    # For OMA: ...
 
-def run_orthology_tools():
-    """Docstring"""
+    Args:
+        source_dir: Path to the directory containing fasta files.
+        target_dir_ofinder: Path to the directory where symlinks to fasta files will be created
+            for OrthoFinder to use.
+
+    Raises:
+        FileExistsError: If directory `target_dir_ofinder` already exists.
+        FileNotFoundError: If directory `source_dir` does not exist or does not contain any fasta files.
+        subprocess.CalledProcessError: If creating symlinks fails for some other reason.
+    """
+    if not os.path.isdir(source_dir):
+        raise FileNotFoundError("Directory containing fasta files not found.")
+    if len(fnmatch.filter(os.listdir(source_dir), '*.fasta')) == 0:
+        raise FileNotFoundError("No fasta files found.")
+
+    # OrthoFinder
+    os.mkdir(target_dir_ofinder)
+    script_symlinks = os.path.join(Path(__file__).parents[2], "scripts", "pipeline", "symlink_fasta.py")
+    subprocess.run([script_symlinks, "-s", target_dir_ofinder, "-d", source_dir],
+                   capture_output=True, check=True)
+
+
+def run_orthology_tools(input_ofinder: str) -> None:
+    """Runs selected orthology inference tools: OrthoFinder.
+
+    Args:
+        input_ofinder: Path to the directory containing symlinks to the input fasta files for OrthoFinder.
+
+    Raises:
+        FileNotFoundError: If OrthoFinder executable (`orthofinder_exe`) cannot be found.
+        subprocess.CalledProcessError: If executing OrthoFinder command fails for some reason,
+            including `input_ofinder` not found.
+    """
+    # OrthoFinder
+    orthofinder_exe = "/hps/software/users/ensembl/ensw/C8-MAR21-sandybridge/linuxbrew/bin/orthofinder"
+    subprocess.run([orthofinder_exe, "-t", str(32), "-a", str(8), "-f", input_ofinder],
+                   capture_output=True, check=True)
```

Comment:
```suggestion
    s_threads = str(32)
    a_threads = str(8)
    subprocess.run([orthofinder_exe, "-t", s_threads, "-a", a_threads, "-f", input_ofinder],
                   capture_output=True, check=True)
```
I suggest this a little for transparency and clarity on what the parameters are pertaining (for those that know nothing about orthofinder), although I think what you have is also fine

Reply:
In fact, I had `number_of_threads` and `number_of_orthofinder_threads` in that abandoned [PR](https://github.com/Ensembl/ensembl-compara/pull/365) but JA was very critical of that:
> Futhermore, I would suggest you to avoid parameter renaming when wrapping external tools, to avoid confusion on usage for users.

Truth be told, the authors use `number_of_threads` and `number_of_orthofinder_threads` in [the manual](https://github.com/davidemms/OrthoFinder/blob/master/OrthoFinder-manual.pdf). 

Since here I don't provide an option to actually pass the number of threads via command line, and `s_threads` is not much less cryptic than `s` (same for `a`) coupled with adding unnecessary variable declarations, I will actually add options to pass these numbers  via command line (with let's say `-t 32 -a 8` as default).

Related to what JA said, what do you think about variable names?

Reply:
The author's also use `search_threads` and `analysis_threads` :point_left: so these may be even more appropriate. This is not renaming the parameters, because the `-t` and `-a` flags remain in the code and have not been renamed, it is the values that are being named to accompany the flags in my suggestion. As parameters that are optimised for a specific environment setup, I would encourage providing them as default values, but also give flexibility to changing them as you mention in the commandline. For this dataset of just 2 genomes, these parameters could even be overkill, they are selected for a dataset of ~40 genomes. But as I said, I think it is fine without, it was just a suggestion :shrug:

Reply:
I didn't rename `-t` and `a`. I had `"-t", str(number_of_threads), "-a", str(number_of_orthofinder_threads)` 😉 

I like `search_threads` and `analysis_threads` 👍 The pipeline is being developed for any number of genomes, and as I said in the description, this is a small test collection to facilitate full pipeline run. Actually, I could've kept those changes in the `mlss_conf.xml` locally 🤔

Reply:
Yes, exactly, I think you are on the same page as me.

Reply:
Errr... searching the manual didn't give any results for `search_threads` and `analysis_threads`. I didn't manage to spot them by scrolling either.
I guess they have switched to `number_of_threads` and `number_of_orthofinder_threads`. `number_of_threads` also sounds generic enough if shared across different tools. So I went with them.

Reply:
```
$ orthofinder --help

...

OPTIONS:
 -t <int>        Number of parallel sequence search threads [Default = 16]
 -a <int>        Number of parallel analysis threads
```

Reply:
Oh, I see what you meant. I was referring to `search_threads`, `analysis_threads`, `number_of_threads`, `number_of_orthofinder_threads`.